src/ConstructiveSolidGeometry/VolumePrimitives/Cone.jl

```diff
@@ -73,13 +73,54 @@ from 2 at the bottom to 4 at the top circular base.
     
 See also [Constructive Solid Geometry (CSG)](@ref).
 """
-@with_kw struct Cone{T,CO,TR,TP<:Union{Nothing,T}} <: AbstractVolumePrimitive{T, CO}
-    r::TR = 1
-    φ::TP = nothing
-    hZ::T = 1
+struct Cone{T,CO,TR,TP<:Union{Nothing,T}} <: AbstractVolumePrimitive{T, CO}
+    r::TR 
+    φ::TP
+    hZ::T 
 
-    origin::CartesianPoint{T} = zero(CartesianPoint{T})
-    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+    origin::CartesianPoint{T} 
+    rotation::SMatrix{3,3,T,9}
+end
+
+#Type conversion happens here
+function Cone{T,CO}(r, φ, hZ, origin, rotation) where {T,CO}
+    _r = _csg_convert_args(T, r)
+    _φ = _csg_convert_args(T, φ)
+    _hZ = _csg_convert_args(T, hZ)
+    Cone{T,CO,typeof(_r),typeof(_φ)}(_r, _φ, _hZ, origin, rotation)
+end
+
+#Type promotion happens here
+function Cone(CO, r::TR, φ::TP, hZ::TZ, origin::PT, rotation::ROT) where {TR, TP<:Real, TZ, PT, ROT}
+    eltypes = _csg_get_promoted_eltype.((TR, TZ, TP, PT, ROT))
+    T = promote_type(eltypes...)
+    Cone{T,CO}(r, φ, hZ, origin, rotation)
+end
+
+function Cone(CO, r::TR, φ::Nothing, hZ::TZ, origin::PT, rotation::ROT) where {TR, TZ, PT, ROT}
+    eltypes = _csg_get_promoted_eltype.((TR, TZ, PT, ROT))
+    T = promote_type(eltypes...)
+    Cone{T,CO}(r, φ, hZ, origin, rotation)
+end
+
+function Cone(::Type{CO}=ClosedPrimitive;
+    r = 1.0, 
+    φ = nothing,
+    hZ = 1.0,
+    origin = zero(CartesianPoint{Float64}), 
+    rotation = one(SMatrix{3, 3, Float64, 9})
+) where {CO}
+    Cone(CO, r, φ, hZ, origin, rotation)
+end
```

Comment:
By defining `r = 1.0`, `hZ = 1.0` and `Float64` and element type of both `origin` and `rotation`, the promoted `Float32` of `tuple` will be ignored as `hZ`, `origin` and `rotation` already require `Float64`.

This could be omitted by using integers here, that do not influence the type promotion, i.e.
```julia
function Cone(::Type{CO}=ClosedPrimitive;
    r = 1, 
    φ = nothing,
    hZ = 1,
    origin = zero(CartesianPoint{Int64}), 
    rotation = one(SMatrix{3, 3, Int64, 9})
) where {CO}
    Cone(CO, r, φ, hZ, origin, rotation)
end
```
...

Reply:
Is there any reason or disadvantage **NOT** to use `Int64` in this constructor, such that
```julia
CSG.Cone(r = tuple)
```
results in a `Cone` of type `Float32`?